app/workers/sphinx_indexation_worker.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+
+# SphinxIndexationWorker updates sphinx index for the provided model.
+# It is enqueued when:
+# - Account gets created, updated (deletion is done automatically as per https://freelancing-gods.com/thinking-sphinx/v3/real_time.html
+# - User gets created, updated, deleted
+# - Cinstance gets created, updated, deleted
+class SphinxIndexationWorker < ApplicationJob
+
+  rescue_from(ActiveJob::DeserializationError) do |exception|
```

Comment:
In which case can this happen? As far as I understood, the perform is for the model, and in what moment can it not exist anymore?

Reply:
Imagine you update the model, sidekiq workers are down, then some process deletes the record, and sidekiq up again
It will be a GlobalId deserialization error

Reply:
ok then when this is legitimately called for a Cinstance/User that has been deleted, can we be sure that it will always be called using [the missing model](https://github.com/3scale/porta/blob/acc77228aa4f1d1ac226c78cae0e421588de6990/app/lib/missing_model.rb#L4-L26) or any other way where we won't have this DeserializationError as a real bug for those cases?

Reply:
The only params sent is either `Account`, `Topic` or `CMS::Page` so no worries.

It will not be called using missing model.
When it gets destroyed ThinkingSphinx automatically remove the record from the index https://freelancing-gods.com/thinking-sphinx/v3/real_time.html#model-callbacks